Sources/Features/DappInteractionFeature/Interactor/DappInteractor.swift

```diff
@@ -333,13 +343,15 @@ struct DappInteractor: Sendable, FeatureReducer {
 							reason: error.localizedDescription
 						)
 					))
+				} else {
+					loggerGlobal.notice("Failed to send response back to dapp, error: \(error), not delegating `sentResponseToDapp`.")
 				}
 			}
 		}
 	}
 
 	func dismissCurrentModalAndRequest(_ request: RequestEnvelope, for state: inout State) {
-		state.requestQueue.remove(request)
+		state.requestQueue.remove(id: request.id)
```

Comment:
The bug was here, we actually never removed the `request` since Equality treated Manifest `M != M`.